lib/generic/states.rb

```diff
@@ -284,11 +284,21 @@ def perform_encounter(time, entity, record_encounter = true)
             cond = entity[@reason].to_sym if cond.nil? && entity[@reason]
           end
 
+          # find closest service provider
+          service = @encounter_class
+          provider = Synthea::Provider.find_closest_service(entity, service)
```

Comment:
Seems like you don't need this, because you already assigned providers to each Entity and Person...